engine/src/retrier.rs

```diff
@@ -405,7 +405,13 @@ where
 							// We avoid small delays by always having a time of at least half.
 							let half_max = max_sleep_duration(initial_request_timeout, attempt) / 2;
 							let sleep_duration = half_max + rand::thread_rng().gen_range(Duration::default()..half_max);
-							tracing::error!("Retrier {name}: Error for request `{request_log}` with id `{request_id}`, attempt `{attempt}`: {e}. Delaying for {}ms", sleep_duration.as_millis());
+
+							let error_message = format!("Retrier {name}: Error for request `{request_log}` with id `{request_id}`, attempt `{attempt}`: {e}. Delaying for {}ms", sleep_duration.as_millis());
+							if attempt == 0 && matches!(retry_limit, RetryLimit::NoLimit) {
```

Comment:
Would this be better as, this is attempt 0 and attempt 0 is not the last attempt?

Reply:
true, think this is a nice improvement, pushed the change